test/multipart_test.gleam

```diff
@@ -0,0 +1,78 @@
+import bucket/complete_multipart_upload
+import bucket/create_multipart_upload
+import bucket/get_object
+import bucket/upload_part
+import gleam/bit_array
+import gleam/httpc
+import gleam/int
+import gleam/list
+import gleam/string
+import gleeunit/should
+import helpers
+
+pub fn perform_multipart_upload_test() {
+  // Set up a test bucket
+  helpers.delete_existing_buckets()
+  let bucket = "multipart1"
+  helpers.create_bucket(bucket)
+
+  // Initiate a multipart upload
+  let key = "test/myfile"
+  let assert Ok(res) =
+    create_multipart_upload.request(bucket:, key:)
+    |> create_multipart_upload.build(helpers.creds)
+    |> httpc.send_bits
+  let assert Ok(res) = create_multipart_upload.response(res)
+  should.equal(res.bucket, bucket)
+  should.equal(res.key, key)
+  should.not_equal(res.upload_id, "")
+
+  // Upload some parts (they can be sent in any order)
+  let upload_id = res.upload_id
+  // NOTE: The minimum Part size for multipart upload is 5MiB (hardcoded in minio)
+  let megabyte = 1024 * 1024
+  let parts =
+    [
+      #(string.repeat("B", 5 * megabyte) |> bit_array.from_string, 2),
+      #(string.repeat("A", 5 * megabyte) |> bit_array.from_string, 1),
+      #(<<"Goodbye!":utf8>>, 3),
+    ]
+    |> list.map(fn(pair) {
+      let #(body, part_number) = pair
+      let assert Ok(res) =
+        upload_part.request(bucket:, key:, upload_id:, part_number:, body:)
```

Comment:
should I actually perform these uploads in parallel so that we have an example use case?